File: versus/go/05-resource-management/resource.go
package main

import (
	"sync"
)

type Stats struct {
	someStat  uint64
	otherStat uint64

	sync.Mutex
}

func updateSharedObject(obj *Stats) {
	obj.Lock()
	defer obj.Unlock()

	obj.someStat++
	obj.otherStat = obj.someStat / 2
}

File: versus/go/05-resource-management/resource.cc
#include <mutex>

struct Stats {
    uint64_t someStat;
    uint64_t otherStat;

    std::mutex lock;
};

void updateSharedObject(Stats *obj) {
    std::lock_guard m(obj->lock);

    obj->someStat++;
    obj->otherStat = obj->someStat / 2;
}
